### PR TITLE
Fix MUNICIPAL user organization assignment and endorse button functionality

### PR DESCRIPTION
**Pull request by camwalford**
This changes a few things and should have just been multiple branches, but 🤡 <- me

**Properly setting organizationName/municipality**
- Municipal seg admins now have their organizationName set to whatever their home segment name is. 
- All municipal users created by a Municipal Segment Admin will have their organizationName set to the admin's organizationName, or the admin's home segment name as a fallback
- All municipal users created by a Super admin or Admin will have their organizationName set to the home segment name selected by the admin in the creation wizard

**Endorsements**
- Restricted endorsement button so only one member of an organization can endorse a proposal or idea. 
- Added tooltip to the disabled endorse button notifying user that the idea has been endorsed by their org already
- Prevented unendorsement of an idea/proposal except by the user who endorsed it (may need to be changed in the future)

### CODE REVIEW

**Comment by camfung** [comment]
lgmt

File: frontend/src/components/content/UserManagementContent.tsx [hunk 1]
```diff
@@ -167,21 +167,43 @@ export const UserManagementContent: React.FC<UserManagementContentProps> = ({ us
...
+        user: IUser | null, 
+        selectedUserType: string, 
+        userVerbose: IUser | null | undefined
+      ): string | undefined => {
+        // Check if the user is a SUPER_ADMIN, ADMIN, or MUNICIPAL_SEG_ADMIN
+        if (user?.userType === USER_TYPES.SUPER_ADMIN || 
+            user?.userType === USER_TYPES.ADMIN || 
+            user?.userType === USER_TYPES.MUNICIPAL_SEG_ADMIN) {
+      
+          // If selected user type is MUNICIPAL or parent is MUNICIPAL_SEG_ADMIN
```
Comment (camfung) [inline frontend/src/components/content/UserManagementContent.tsx]: this comment shouldnt be all caps

File: frontend/src/components/content/UserManagementContent.tsx [hunk 2]
```diff
@@ -167,21 +167,43 @@ export const UserManagementContent: React.FC<UserManagementContentProps> = ({ us
...
+      
     const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
         event.preventDefault();
         const form = event.target as HTMLFormElement;
         const formData = new FormData(form);
-
+        
+        const orgName = getOrganizationName(user, selectedUserType, userVerbose) ?? 
+                        (selectedUserType === USER_TYPES.BUSINESS || selectedUserType === USER_TYPES.COMMUNITY) ? 
+                        formData.get('inputOrg') as string : undefined;
```
Comment (camfung) [inline frontend/src/components/content/UserManagementContent.tsx]: Make sure you're using the correct logical operaotrs

Comment (camwalford) [reply frontend/src/components/content/UserManagementContent.tsx]: 👌

**Pull request merged**